app/views/handlers/submission.py

```diff
@@ -13,14 +13,18 @@ def __init__(self, schema, questionnaire_store, full_routing_path):
         self._schema = schema
         self._questionnaire_store = questionnaire_store
         self._full_routing_path = full_routing_path
+        self.session_store = get_session_store()
 
     def submit_questionnaire(self):
-        message = json.dumps(
-            convert_answers(
-                self._schema, self._questionnaire_store, self._full_routing_path
-            ),
-            for_json=True,
+
+        payload = convert_answers(
+            self._schema, self._questionnaire_store, self._full_routing_path
         )
+        payload[
```

Comment:
this can be done inside the `convert_answers` method

Reply:
convert_answers  doesn't feel like the right location, as it's used by both flusher and dumper as well. In those 2 instances language_submission_code should not be present in the payload and therefore having it there would need extra logic for 2 out of 3 situations.

Reply:
Just saw your previous comment 🤦